rust/src/lib.rs

```diff
@@ -0,0 +1,17 @@
+use pyo3::prelude::*;
```

Comment:
(Somewhat leaving this comment in a random spot...)

Will integrating Rust code this way end up with the same issues we ran into on #10857 -- which I think boils down to "what do we do about metrics in Rust code?"

Reply:
Yes, though we have that problem today with other native modules. The scale of the problem however is different: the state compressor happens in the background and consumed multiple cores, while most of the native code we run is "inline" on the main thread and doesn't consume much CPU.

We probably will want to figure something out here if we end up doing lots of CPU in rust, but I don't see it as blocking.

Reply:
That makes sense, thanks for the explanation!